lmfdb/classical_modular_forms/test_cmf.py

```diff
@@ -937,4 +909,53 @@ def test_is_self_dual(self):
         for elt in ['52.1.j.a', '57.1.h.a', '111.1.h.a']:
             assert elt in page.get_data(as_text=True)
 
+    def test_hecke_charpolys(self):
+        """Test that the Hecke charpolys are correct.
+
+        Some expected Hecke charpolys are stored in the dict test_data,
+        which are then checked to be in the relevant page. These examples
+        have been chosen to be readily verifiable from the displayed
+        Fourier coefficients of each respective homepage."""
+
+        test_data = {
+                    '11/2/a/a' : {2: '\( 2 + T \)',
+                                 17: '\( 2 + T \)',
+                                 29: '\( T \)'},
+
+                    '10/3/c/a' :  {5: r'\( 25 + T^{2} \)',
+                                  11: r'\( ( 8 + T )^{2} \)',
+                                  97: r'\( 7938 + 126 T + T^{2} \)'},
+
+                    '/294/5/b/f' : {2: r'\( ( 8 + T^{2} )^{5} \)',
+                                    # The following test checks that monomials do not have superfluous parentheses
+                                    7: r'\( T^{10} \)'},
+
+                    # The following is a newspace with 5 newforms, hence the more verbose polynomials.
+                    # These newspace examples have historically also been prone to superfluous parentheses
+                    # (e.g. LMFDB Issue #4117)
+
+                    '255/1/h/' : {2: r'(\( 1 + T \))(\( 1 + T \))(\( -1 + T \))(\( -1 + T \))(\( T^{2} \))',
+                                  3: r'(\( 1 + T \))(\( -1 + T \))(\( 1 + T \))(\( -1 + T \))(\( 1 + T^{2} \))',
+                                  5: r'(\( 1 + T \))(\( -1 + T \))(\( -1 + T \))(\( 1 + T \))(\( 1 + T^{2} \))'},
+
+                    # The following is a newspace with 9 newforms, but only 7 have been computed exactly.
```

Comment:
Should we even be displaying Hecke Characteristic Polynomials in this case? There are 9 newforms, but only 7 are computed exactly; for the remaining two newforms, the characteristic polynomials are not in the database. So the displayed product of 7 polynomials is only a *factor* of the true Hecke Characteristic polynomial of this newspace, and not the whole thing? If so, either we make clear that "The following listed polynomials are only factors", or we say "The polynomials are not in the database", because some factors are missing?

Reply:
No, we should only show charpolys for spaces where we actually know what it is (which for newspaces means we need to know the charpolys for all the newforms in the space).

Reply:
This will be addressed by not showing Hecke charpolys for newspaces.